### ISSUE TITLE
Only update specific apps

### ISSUE THREAD

**Issue by KnifMelti** [Romanitho/Winget-AutoUpdate#36]
As an alternativ to the 'excluded_apps.txt' it would be nice to have a 'included_apps.txt' for whitelisting purpose, i.e when specifik apps are installed via SCCM that list could be updated and WAU should only update these, not anything else.

**Comment by Romanitho** [comment]
Hi.
Thanks for your comment. It was in my mind, so thanks to have written it down.
Just thinking about the best way to have white and black list options in install script.

**Comment by KnifMelti** [comment]
My take on this is:
If there exist a white list (populated or unpopulated) then nothing gets updated **until** the white list gets populated.
I'm saving all our apps that we let the users manage as **Available** in SCCM in registry as Detection Method under HKEY_LOCAL_MACHINE\SOFTWARE\Winget
InstEd.InstEd=Installed
TimKosse.FileZilla.Client=Uninstalled
etc.

If/when there's a whitelist for WAU I'm gonna populate the list from the registry when users install/uninstall an app from SCCM

**Comment by Romanitho** [comment]
[v1.7.0](https://github.com/Romanitho/Winget-AutoUpdate/releases/tag/v1.7.0) published as pre-release for testing.

You can now use white list to only maintain app you want.

Juste create an included_apps.txt next to excluded_apps.txt, add the apps in list and run ".\Winget-AutoUpdate-Install.ps1 -UseWhiteList"

**Comment by Romanitho** [comment]
The idea if you use White List is :
if a file exists, keep it
if no file, create empty one

And if you use [Winget-Install,](https://github.com/Romanitho/Winget-Install) you can install apps and use the "-WAUWhiteList" switch to add the app to white list when it's installed

**Comment by KnifMelti** [comment]
Wonderful!
Used your WAU (stripped version) for install/repair (update)/uninstall single apps in SCCM before (because of the non existing whitelist feature in WAU.
In that way users could manually update single apps from Software Center.
Going to abandon my own additions and begin using your Winget-Install + WAU and $WAUWhiteList instead!
Thanks!

**Comment by Romanitho** [comment]
glad it helped ;)

**Comment by kylejericson** [comment]
Need some help with the whitelist file. I'm installing this via Intune and I wrapped my whitelist file inside the win32 app file. I seem to be getting this error. How should I fix existing devices and my new deployments via Intune.


![Screenshot-08-13-01-AM](https://github.com/Romanitho/Winget-AutoUpdate/assets/43736908/1549acee-fe04-4081-b7a2-cb8fff27b97f)


![Screenshot-08-13-16-AM](https://github.com/Romanitho/Winget-AutoUpdate/assets/43736908/21b19a13-1ce2-42bd-9df2-a4e0e097cb17)

**Comment by KnifMelti** [comment]
Hi!
Don't include ‐Listpath in your install. It will find the list itself (default).

**Comment by kylejericson** [comment]
Ok so for those that were deployed with the ‐Listpath is there a way to fix them after the fact?

**Comment by Romanitho** [comment]
Yes, simply redeploy WAU.

**Comment by kylejericson** [comment]
Ok did that and same issue.

**Comment by kylejericson** [comment]
Does this look like it's working?

![Screenshot-11-47-18-AM](https://github.com/Romanitho/Winget-AutoUpdate/assets/43736908/b5d0fdd8-c813-4c8f-a93c-b22947a97d14)

**Comment by kylejericson** [comment]
My folder contents:

![Screenshot-11-47-49-AM](https://github.com/Romanitho/Winget-AutoUpdate/assets/43736908/6c1a5cc4-b50d-4ec0-b731-f7b28c6e0e46)

**Comment by KnifMelti** [comment]
Strange then...
...there's the string of dashes (when there's not the message of An unusual... pops up).
Are those apps in your white list?

**Comment by kylejericson** [comment]
No this is my whitelist.

Google.Chrome
AgileBits.1Password
Perimeter81Ltd.Perimeter81Ltd
SlackTechnologies.Slack
Zoom.Zoom

**Comment by KnifMelti** [comment]
Well, run the same command as before as system in a terminal (psexec or nirsoft advanced run) and post the output.

**Comment by KnifMelti** [comment]
https://www.nirsoft.net/utils/advanced_run.html

**Comment by Romanitho** [comment]
I agree, in your case we need to see what winget returns in system context.

**Comment by KnifMelti** [comment]
Or/and the log from after (**$upgradeResult** is transferred to the log):
![image](https://github.com/Romanitho/Winget-AutoUpdate/assets/102996177/ef38d5d4-6771-46f7-9c32-05079aeae71e)
